feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/mappers/Validator.kt

```diff
@@ -127,27 +129,44 @@ fun mapValidatorToValidatorDetailsWithStakeFlagParcelModel(
         val stakeModel = electedInfo?.let {
             val nominators = it.nominatorStakes.map(::mapNominatorToNominatorParcelModel)
 
-            ValidatorStakeParcelModel.Active(it.totalStake, it.ownStake, nominators, it.apy, slashed, it.maxNominators, isNominated)
+            val indexOfCurrentAccount = nominators.sortedBy { it.value }.indexOfFirst { it.who.contentEquals(account.address.toAccountId()) }
+            val isInLimit = indexOfCurrentAccount < it.maxNominators
+
+            val isOversubscribed = nominators.size > it.maxNominators
+
+            ValidatorStakeParcelModel.Active(
+                it.totalStake,
+                it.ownStake,
+                nominators,
+                it.apy,
+                slashed,
+                ValidatorStakeParcelModel.Active.NominatorInfo(
+                    it.maxNominators,
+                    isNominated,
+                    isInLimit,
+                    isOversubscribed
+                )
+            )
         } ?: ValidatorStakeParcelModel.Inactive
 
         ValidatorDetailsParcelModel(accountIdHex, stakeModel, identityModel)
     }
 }
 
 fun mapValidatorDetailsToErrors(
-    validator: ValidatorDetailsParcelModel,
-    account: StakingAccount
+    validator: ValidatorDetailsParcelModel
 ): Error? {
     return when (val stake = validator.stake) {
         ValidatorStakeParcelModel.Inactive -> null
         is ValidatorStakeParcelModel.Active -> {
-            when {
-                stake.isOversubscribed() && !stake.isNominated -> Error.OversubscribedPaid
-                stake.isOversubscribed() && stake.isInLimit(account.address) -> Error.OversubscribedPaid
-                stake.isOversubscribed() && !stake.isInLimit(account.address) -> Error.OversubscribedUnpaid
-                stake.isSlashed -> Error.Slashed
-                else -> null
-            }
+            val nominatorInfo = stake.nominatorInfo ?: return null
+
+            return if (nominatorInfo.isOversubscribed) {
+                if (!nominatorInfo.isNominated) Error.OversubscribedPaid
+                else if (nominatorInfo.isInLimit) Error.OversubscribedPaid
+                else Error.OversubscribedUnpaid
+            } else if (stake.isSlashed) Error.Slashed
+            else null
```

Comment:
Errors are not mutually exclusive. Meaning, slashing may occur alongside with oversubscription.

I think the correct way is to return List<Error> here filling it something like
```
return mutableListOf().apply {
 // add() to add error
}
```

Reply:
Done

Reply:
Done